docs/advanced.md

```diff
@@ -68,79 +68,68 @@ Additionally, `Client` accepts some parameters that aren't available at the requ
 One particularly useful parameter is `base_url`, which allows you to define a base URL to prepend to all outgoing requests:
 
 ```python
->>> async with httpx.AsyncClient(base_url='http://httpbin.org') as client:
-...     r = await client.get('/headers')
+>>> with httpx.Client(base_url='http://httpbin.org') as client:
+...     r = client.get('/headers')
 ...
 >>> r.request.url
 URL('http://httpbin.org/headers')
 ```
 
-For a list of all available client-level parameters, see the [`AsyncClient` API reference](/api/#async-client).
+For a list of all available client-level parameters, see the [`Client` API reference](/api/#client).
 
 ## Calling into Python Web Apps
 
-You can configure an `httpx` client to call directly into a Python web
-application using the ASGI protocol.
+You can configure an `httpx` client to call directly into a Python web application using the WSGI protocol.
 
 This is particularly useful for two main use-cases:
 
 * Using `httpx` as a client inside test cases.
 * Mocking out external services during tests or in dev/staging environments.
 
-Let's take this Starlette application as an example:
+Here's an example of integrating against a Flask application:
 
 ```python
-from starlette.applications import Starlette
-from starlette.responses import HTMLResponse
-from starlette.routing import Route
-
+from flask import Flask
+import httpx
 
-async def hello():
-    return HTMLResponse("Hello World!")
 
+app = Flask(__name__)
 
-app = Starlette(routes=[Route("/", hello)])
-```
+@app.route("/")
+def hello():
+    return "Hello World!"
 
-We can make requests directly against the application, like so:
-
-```python
->>> import httpx
->>> async with httpx.AsyncClient(app=app) as client:
-...     r = client.get('http://example/')
-...     assert r.status_code == 200
-...     assert r.text == "Hello World!"
+with httpx.Client(app=app) as client:
+    r = client.get('http://example/')
+    assert r.status_code == 200
+    assert r.text == "Hello World!"
 ```
 
-For some more complex cases you might need to customise the ASGI dispatch. This allows you to:
+For some more complex cases you might need to customize the WSGI dispatch. This allows you to:
 
 * Inspect 500 error responses rather than raise exceptions by setting `raise_app_exceptions=False`.
-* Mount the ASGI application at a subpath by setting `root_path`.
-* Use a given client address for requests by setting `client`.
+* Mount the WSGI application at a subpath by setting `script_name` (WSGI).
+* Use a given client address for requests by setting `remote_addr` (WSGI).
 
 For example:
 
 ```python
-# Instantiate a client that makes ASGI requests with a client IP of "1.2.3.4",
-# on port 123.
-dispatch = httpx.dispatch.ASGIDispatch(app=app, client=("1.2.3.4", 123))
-async with httpx.AsyncClient(dispatch=dispatch) as client:
+# Instantiate a client that makes WSGI requests with a client IP of "1.2.3.4".
+dispatch = httpx.dispatch.WSGIDispatch(app=app, remote_addr="1.2.3.4")
```

Comment:
Would we want that to become `httpx.WSGIDispatch`?

Reply:
Good catch, yes. Didn't notice the `.dispatch.` that crept in.